scipy/interpolate/interpolate.py

```diff
@@ -486,6 +488,19 @@ def __init__(self, x, y, kind='linear', axis=-1,
                 self.x_bds = self.x_bds[1:] + self.x_bds[:-1]
 
                 self._call = self.__class__._call_nearest
+            elif kind == 'previous':
+                # Side for np.searchsorted and index for clipping
+                self.side = 'left'
```

Comment:
Please add a leading underscore to private attributes, i.e., self._side and self._ind